File: Emulator/src/main/core/game/world/map/zone/ZoneRestriction.java
package core.game.world.map.zone;

/**
 * The enum Zone restriction.
 */
public enum ZoneRestriction {

    /**
     * Followers zone restriction.
     */
    FOLLOWERS,

    /**
     * Random events zone restriction.
     */
    RANDOM_EVENTS,

    /**
     * Fires zone restriction.
     */
    FIRES,

    /**
     * Members zone restriction.
     */
    MEMBERS,

    /**
     * Cannon zone restriction.
     */
    CANNON,

    /**
     * Graves zone restriction.
     */
    GRAVES,

    /**
     * Teleport zone restriction.
     */
    TELEPORT;

    /**
     * Gets flag.
     *
     * @return the flag
     */
    public int getFlag() {
        return 1 << ordinal();
    }
}

File: Server/src/main/core/game/world/map/zone/ZoneRestriction.kt
package core.game.world.map.zone

/**
 * The zone restrictions.
 *
 * @author Emperor
 */
enum class ZoneRestriction {
    FOLLOWERS,          // No followers allowed in this zone.
    RANDOM_EVENTS,      // No random events allowed.
    FIRES,              // No fires allowed.
    MEMBERS,            // Members only.
    CANNON,             // No cannons allowed.
    GRAVES,             // Do not spawn a grave if a player dies here.
    TELEPORT,           // No teleporting allowed.
    ;

    val flag: Int
        /**
         * Gets the restriction flag.
         *
         * @return The flag.
         */
        get() = 1 shl ordinal
}
